### ISSUE TITLE
By.LinkText doesn't work with elements having text-transform:capitalize style

### ISSUE THREAD

**Issue by romovs** [SeleniumHQ/selenium#884]
Selenium fails to find elements which have "text-transform:capitalize" style applied.
Please see the test case below. Selenium finds elements which have "uppercase" and "lowercase" or no transformations successfully, but fails on "capitalize".

Tested with Selenium v2.45.0 (C#). Chrome/FF, Windows/Linux

Example html:

``` html
<html>
    <head></head>
    <body>
        <a href="#" id="upper" style="text-transform:uppercase;">Hello world! foo-bar-BAZ</a><br/>
        <a href="#" id="lower" style="text-transform:lowercase;">Hello world! foo-bar-BAZ</a><br/>
        <a href="#" id="capital" style="text-transform:capitalize;">Hello world! foo-bar-BAZ</a><br/>
        <a href="#" id="none">Hello world! foo-bar-BAZ</a>
    </body>
</html>
```

The transformed text looks like this:

```
HELLO WORLD! FOO-BAR-BAZ
hello world! foo-bar-baz
Hello World! Foo-Bar-BAZ
Hello world! foo-bar-BAZ
```

Example test:

``` cs
using OpenQA.Selenium;
using OpenQA.Selenium.Chrome;

namespace SeleniumLinkTest
{
    class Program
    {
        static void Main(string[] args)
        {
            using (var driver = new ChromeDriver())
            {
                driver.Navigate().GoToUrl("http://localhost");

                var upper = driver.FindElement(By.LinkText("HELLO WORLD! FOO-BAR-BAZ"));
                var lower = driver.FindElement(By.LinkText("hello world! foo-bar-baz"));
                var capital = driver.FindElement(By.LinkText("Hello World! Foo-Bar-BAZ")); // fails with NoSuchElementException
                var none = driver.FindElement(By.LinkText("Hello world! foo-bar-BAZ"));
            }
        }
    }
}
```

**Comment by jleyba** [comment]
Problem appears to be that Chrome considers `-` a word boundary while our JS library does not:
https://github.com/SeleniumHQ/selenium/blob/master/javascript/atoms/dom.js#L1293

Firefox does not treat `-` as a word boundary: http://jsfiddle.net/2j8dnzz5/1/

**Comment by AutomatedTester** [comment]
Unfortunately there is no defined behaviour here, as with a few other places, in the CSS Specification. This is going to be extremely difficult to recreate in JS since we dont have the ability to create word boundaries like the browser would... at least not easily.

**Comment by SimonSapin** [comment]
CSS leaves the exact behavior of `capitalize` undefined:

https://drafts.csswg.org/css-text/#text-transform-property
> For capitalize, what constitutes a “word“ is UA-dependent; [UAX29] is suggested (but not required) for determining such word boundaries.

https://w3c.github.io/webdriver/webdriver-spec.html#link-text refers to "visible text" but does not define what that is. Who says that should be affected by `text-transform`? Given the above, that doesn’t seem like a great idea.

**Comment by barancev** [comment]
Pushing this issue to chromedriver side:
https://bugs.chromium.org/p/chromedriver/issues/detail?id=2155

Should we fix the atom and propagate updated atom version to chromedriver?

**Comment by barancev** [comment]
Fixed by commit c0f3e3c743651b9817dbe69f7ba7614e86eda91c

At least we'll capitalize the character that follows what a browser thinks is a word boundary (\b in a regexp)
